packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -437,105 +477,100 @@ export class TokenDetectionController extends StaticIntervalPollingController<
 
     const { allTokens, allDetectedTokens, allIgnoredTokens } =
       this.messagingSystem.call('TokensController:getState');
-    const tokens = allTokens[chainId]?.[selectedAddress] ?? [];
-    const detectedTokens = allDetectedTokens[chainId]?.[selectedAddress] ?? [];
-    const ignoredTokens = allIgnoredTokens[chainId]?.[selectedAddress] ?? [];
-
+    const [tokensAddresses, detectedTokensAddresses, ignoredTokensAddresses] = [
+      allTokens,
+      allDetectedTokens,
+      allIgnoredTokens,
+    ].map((tokens) =>
+      (
+        tokens[chainIdAgainstWhichToDetect]?.[addressAgainstWhichToDetect] ?? []
+      ).map((value) => (typeof value === 'string' ? value : value.address)),
+    );
     const tokensToDetect: string[] = [];
     for (const tokenAddress of Object.keys(tokenListUsed)) {
       if (
-        !findCaseInsensitiveMatch(
-          tokens.map(({ address }) => address),
-          tokenAddress,
-        ) &&
-        !findCaseInsensitiveMatch(
-          detectedTokens.map(({ address }) => address),
-          tokenAddress,
+        [
+          tokensAddresses,
+          detectedTokensAddresses,
+          ignoredTokensAddresses,
+        ].every(
+          (addresses) =>
+            !addresses.find((address) =>
+              isEqualCaseInsensitive(address, tokenAddress),
+            ),
         )
       ) {
         tokensToDetect.push(tokenAddress);
       }
     }
-    const sliceOfTokensToDetect = [];
-    sliceOfTokensToDetect[0] = tokensToDetect.slice(0, 1000);
-    sliceOfTokensToDetect[1] = tokensToDetect.slice(
+    const slicesOfTokensToDetect = [];
+    slicesOfTokensToDetect[0] = tokensToDetect.slice(0, 1000);
+    slicesOfTokensToDetect[1] = tokensToDetect.slice(
       1000,
       tokensToDetect.length - 1,
     );
-
-    /* istanbul ignore else */
-    if (!selectedAddress) {
-      return;
-    }
-
-    for (const tokensSlice of sliceOfTokensToDetect) {
+    for (const tokensSlice of slicesOfTokensToDetect) {
       if (tokensSlice.length === 0) {
         break;
       }
 
       await safelyExecute(async () => {
         const balances = await this.#getBalancesInSingleCall(
-          selectedAddress,
+          addressAgainstWhichToDetect,
           tokensSlice,
+          networkClientIdAgainstWhichToDetect,
         );
-        const tokensToAdd: Token[] = [];
+        const tokensWithBalance: Token[] = [];
         const eventTokensDetails: string[] = [];
-        let ignored;
-        for (const tokenAddress of Object.keys(balances)) {
-          if (ignoredTokens.length) {
-            ignored = ignoredTokens.find(
-              (ignoredTokenAddress) =>
-                ignoredTokenAddress === toChecksumHexAddress(tokenAddress),
-            );
-          }
-          const caseInsensitiveTokenKey =
-            findCaseInsensitiveMatch(
-              Object.keys(tokenListUsed),
-              tokenAddress,
-            ) ?? '';
-
-          if (ignored === undefined) {
```

Comment:
This logic is removed in the extension and moved [here](https://github.com/MetaMask/core/pull/3923/files#r1498478712). See https://github.com/MetaMask/metamask-extension/pull/22814/files#diff-323d0cf46498be3850b971474905354ea5ccf7fa13745ad1e6eba59c5b586830R158-R172

Reply:
This is so much easier to read! 😮‍💨